components/dialog/test/dialog.visual-diff.js

```diff
@@ -46,23 +46,38 @@ describe('d2l-dialog', () => {
 						await page.setViewport({ width: info.viewport.width, height: info.viewport.height, deviceScaleFactor: 2 });
 					});
 
-					it('opened', async function() {
-						if (info.category === 'short-narrow') return; // TODO: remove this; skipping for now due to flaking
-						await helper.open(page, '#dialog');
-						await visualDiff.screenshotAndCompare(page, this.test.fullTitle());
+					it('opened', function() {
+						return new Promise((resolve) => {
```

Comment:
Any ideas on a better way to write these? It seems like it could be simplified. I tried with `reduce` but couldn't quite get it.

Reply:
What about something like...

```javascript
return new Promise((resolve) => {
    helper.open(page, '#dialog').then(() => {
        return page.waitForSelector('#bottom', { visible: true });
    }).then(() => {
        return visualDiff.screenshotAndCompare(page, this.test.fullTitle()
    }).then(() => resolve);
});
```

Reply:
I think think it could be written a couple different ways:

```javascript
it('opened', async function() {
    await helper.open(page, '#dialog');
    await page.waitForSelector('#bottom', { visible: true });
    await visualDiff.screenshotAndCompare(page, this.test.fullTitle());
});
```

However, specific to what's currently here, it is worth mentioning that all `async` methods automatically return a `Promise`, and `await` is synonyms with `then`.  So what's currently written could also be rewritten as:

```javascript
it('opened', function() {
    return helper.open(page, '#dialog').then(() => {
        return page.waitForSelector('#bottom', { visible: true });
    }).then(() => {
        return visualDiff.screenshotAndCompare(page, this.test.fullTitle());
    });
});
```

There are various permutations that combine `Promise`, `then`, `async`, `await` that could be written, but normally I would opt for keeping consistent with `async`/`await` or `Promise`/`then`/`catch`/`finally`.  There are some differences however in the way exceptions are handled, and also, if one wants to wait for multiple promises, the Promise API provides other very helpful methods such as `all`, `race`, `allSettled`.

On this topic, something I see people do sometimes which is pointless is:

```javascript
const someMethod = async() => {
    return await someOtherAsyncMethod();
};
// or
const someMethod = async() => {
    await someOtherAsyncMethod();
};

await someMethod();
```

Which could just be:

```javascript
const someMethod = () => {
    return someOtherAsyncMethod();
};

await someMethod();
```